docket/config.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import print_function
+# A central location for the config dictionary
+#   see application for initialization
+import yaml
+import os
+import logging
+
+levels = {
+    'debug'  :logging.DEBUG,
+    'info'   :logging.INFO,
+    'warn'   :logging.WARNING,
+    'warning':logging.WARNING,
+    'error'  :logging.ERROR,
+    'crit'   :logging.CRITICAL,
+}
+log_level = lambda x: levels.get(x, logging.INFO) if type(x) is str else int(x)
+
+class Config:
```

Comment:
I like how you broke this out into a class. We can probably simplify some of the other app initialization. The goal is that we can read from a YAML to override default behavior. We should also be able to read from the environment. Finally, there should probably be some relatively sane defaults in case neither the YAML nor environment are available.

So maybe, in order of execution:
1. Load defaults in class initialization
2. Override with YAML configuration file
3. Override with environment (highest precedence)

The reason is to support deployment in a container with dynamic configuration from kubernetes (that's where we're headed).

Reply:
Currently we only read a few values from the environment (and they do override the config file). I don't know enough about docker / ansible configuration to know how desirable environment configuration is, but I personally prefer finding my configuration explicitly laid out in one file under /etc .. complicated => overlooked / forgotten.

Reply:
New commit - anything in the environment overrides config file, config file overrides defaults.